docs/tutorials/using-qos-ensurance.md

```diff
@@ -166,9 +167,208 @@ spec:
+
+Taking the ranking of CPU usage metric as an example, it also extends some ranking strategies related to its own metric, such as the ranking of CPU usage, which will compare the priority of two pods in turn. If the priority is the same, then compare the CPU consumption. If the CPU consumption is also the same, continue to compare the extended CPU resource consumption, and finally compare the running time of pod, when there is a difference in an indicator, the comparison result can be returned: `orderedby (classandpriority, CpuUsage, extcpuusage, runningtime) Sort(pods)`
+
+2. Refer to the waterline and pod usage to perform avoidance action
+```go
+   //Divide all the metrics that trigger the waterline threshold into two parts according to their quantified attribute
+   metricsQuantified, MetricsNotQuantified := ThrottleDownWaterLine.DivideMetricsByQuantified()
+   // If there is a metric that cannot be quantified, obtain the metric of a throttleable with the highest actionpriority to operate on all selected pods
+   if len(MetricsNotThrottleQuantified) != 0 {
+       highestPrioriyMetric := GetHighestPriorityThrottleAbleMetric()
+       t.throttlePods(ctx, &totalReleased, highestPrioriyMetric)
+   } else {
+       //Get the latest usage, get the gap to waterline
+       ThrottoleDownGapToWaterLines = buildGapToWaterLine(ctx.getStateFunc())
+       //If the real-time consumption of metric in the trigger waterline threshold cannot be obtained, chose the metric which is throttleable with the highest actionpriority to suppress all selected pods
+       if ThrottoleDownGapToWaterLines.HasUsageMissedMetric() {
+           highestPrioriyMetric := ThrottleDownWaterLine.GetHighestPriorityThrottleAbleMetric()
+           errPodKeys = throttlePods(ctx, &totalReleased, highestPrioriyMetric)
+       } else {
+           var released ReleaseResource
+           //Traverse the quantifiable metrics in the metrics that trigger the waterline: if the metric has a sorting method, use its sortfunc to sort the pod directly, 
+		   //otherwise use generalsorter to sort; Then use its corresponding operation method to operate the pod, and calculate the amount of resources released from the corresponding metric until the gap between the corresponding metric and the waterline no longer exists
+           for _, m := range metricsQuantified {
+               if m.SortAble {
+                   m.SortFunc(ThrottleDownPods)
+               } else {
+                   GeneralSorter(ThrottleDownPods)
+               }
+       
+               for !ThrottoleDownGapToWaterLines.TargetGapsRemoved(m) {
+                   for index, _ := range ThrottleDownPods {
+                       released = m.ThrottleFunc(ctx, index, ThrottleDownPods, &totalReleased)
+                       ThrottoleDownGapToWaterLines[m] -= released[m]
+                   }
+               }
+           }
+       }
+   }
+   ```
```

Comment:
Please remove the blank.

<img width="826" alt="image" src="https://user-images.githubusercontent.com/35299017/179460200-88c53e0a-2ce8-4425-aa80-340a53cba77d.png">